.ci/updatecli/updatecli.d/update-gherkin-specs.yml

```diff
@@ -0,0 +1,82 @@
+name: update-gherkin-specs
+pipelineid: update-gherkin-specs
+
+scms:
+  default:
+    kind: github
+    spec:
+      user: '{{ requiredEnv "GIT_USER" }}'
+      email: '{{ requiredEnv "GIT_EMAIL" }}'
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GIT_USER" }}'
+      branch: "{{ .github.branch }}"
+  apm:
+    kind: github
+    spec:
+      user: '{{ requiredEnv "GIT_USER" }}'
+      email: '{{ requiredEnv "GIT_EMAIL" }}'
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.apm_repository }}"
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      username: '{{ requiredEnv "GIT_USER" }}'
+      branch: "{{ .github.branch }}"
+
+sources:
+  sha:
+    kind: file
+    spec:
+      file: 'https://github.com/{{ .github.owner }}/{{ .github.apm_repository }}/commit/{{ .github.branch }}.patch'
+      matchpattern: "^From\\s([0-9a-f]{40})\\s"
+    transformers:
+      - findsubmatch:
+          pattern: "[0-9a-f]{40}"
+  pull_request:
+    kind: shell
+    dependson:
+      - sha
+    spec:
+      command: gh api /repos/{{ .github.owner }}/{{ .github.apm_repository }}/commits/{{ source "sha" }}/pulls --jq '.[].html_url'
+      environments:
+        - name: GITHUB_TOKEN
+        - name: PATH
+  agents-gherkin-specs-tarball:
+    kind: shell
+    scmid: apm
+    dependson:
+      - sha
+    spec:
+      command: tar cvzf {{ requiredEnv "GITHUB_WORKSPACE" }}/gherkin-specs.tgz .
+      environments:
+        - name: PATH
+      workdir: "{{ .specs.apm_gherkin_path }}"
+
+actions:
+  pr:
+    kind: "github/pullrequest"
+    scmid: default
+    name: '[Automation] Update Gherkin specs'
+    spec:
+      automerge: false
+      draft: false
+      labels:
+        - "automation"
+      description: |-
+        ### What
+        APM agent Gherkin specs automatic sync
+
+        ### Why
+        *Changeset*
+        * {{ source "pull_request" }}
+        * https://github.com/elastic/apm/commit/{{ source "sha" }}
+
+targets:
+  agent-gherkin-specs:
+    name: APM agent gherkin specs {{ source "sha" }}
+    scmid: default
+    sourceid: sha
+    kind: shell
+    spec:
+      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/gherkin-specs.tgz && git --no-pager diff # required to ignore the sourceid being appended'
```

Comment:
The tricky thing here is to use `#` since the value of the `sourceid: sha` is appended to the `command` and then produce some unexpected output.

Reply:
I think if you omit the `sourceid` option, the value is not appended

Reply:
> I think if you omit the sourceid option, the value is not appended

That was expected, but It was not the case for some reason. 

Given

```diff
diff --git a/.ci/updatecli/updatecli.d/update-specs.yml b/.ci/updatecli/updatecli.d/update-specs.yml
index 0be8fa56a..ca40c47ef 100644
--- a/.ci/updatecli/updatecli.d/update-specs.yml
+++ b/.ci/updatecli/updatecli.d/update-specs.yml
@@ -77,7 +77,6 @@ targets:
   agent-json-schema:
     name: APM agent json server schema {{ source "sha" }}
     scmid: default
-    sourceid: sha
     kind: shell
     spec:
       command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-schema.tgz && git --no-pager diff # required to ignore the sourceid being appended'

```

When running:

```bash
GITHUB_WORKSPACE=/tmp PATH=/opt/homebrew/bin:$PATH GIT_USER="foo" GIT_EMAIL="foo@gmail.com" GITHUB_TOKEN=$(gh auth token) updatecli apply --config .ci/updatecli/updatecli.d/update-specs.yml --values .ci/updatecli/values.yml --push=false --debug
```

Then it failed with the below error:

```
+++++++++++
+ PREPARE +
+++++++++++

Loading Pipeline ".ci/updatecli/updatecli.d/update-specs.yml"
ERROR: empty 'sourceid' for target "agent-json-schema"
ERROR: Skipping manifest in position '\x00' from ".ci/updatecli/updatecli.d/update-specs.yml": targets validation error:
wrong updatecli configuration
ERROR: failed loading pipeline(s)
	* ".ci/updatecli/updatecli.d/update-specs.yml" - targets validation error:
wrong updatecli configuration

0 pipeline(s) successfully loaded

SCM repository retrieved: 0

++++++++++++++++++
+ AUTO DISCOVERY +
++++++++++++++++++

ERROR: ✗ no valid pipeline found
ERROR: command failed: no valid pipeline found
```

While if I run this PR as is then:

```bash
TARGETS
========

agent-json-schema
-----------------
DEBUG: No shell success criteria defined, updatecli fallbacks to historical workflow
DEBUG: stage: git-checkout

DEBUG: checkout branch "updatecli_update-schema-specs", based on "main" to directory "/var/folders/t7/ghqdh8cx2g12pwb_w0ncmw900000gn/T/updatecli/github/elastic/apm-agent-java"
DEBUG: 
DEBUG: branch 'updatecli_update-schema-specs' doesn't exist, creating it from branch 'main'
DEBUG: branch "updatecli_update-schema-specs" successfully created
DEBUG: 	command: /bin/sh /var/folders/t7/ghqdh8cx2g12pwb_w0ncmw900000gn/T/updatecli/bin/c3007d0e8db2c2ace0ca4104b167b411531c865a1844a662c648deeecede55b8.sh
DEBUG: Environment variables
DEBUG: 	* UPDATECLI_PIPELINE_STAGE=target
DEBUG: 	* DRY_RUN=false
The shell 🐚 command "/bin/sh /var/folders/t7/ghqdh8cx2g12pwb_w0ncmw900000gn/T/updatecli/bin/c3007d0e8db2c2ace0ca4104b167b411531c865a1844a662c648deeecede55b8.sh" ran successfully with the following output:
----
----
DEBUG: command stderr output was:
----
----
No change detected
✔ - ran shell command "tar -xzf /tmp/json-schema.tgz && git --no-pager diff # required to ignore the sourceid being appended 80b6af8d97e18344df38968bc5b9588961e89492"
DEBUG: Checking if local changes have been done that should be published
DEBUG: no changes detected between branches "main" and "updatecli_update-schema-specs"

ACTIONS
========
=============================

REPORTS:

✔ update-specs:
	Source:
		✔ [agent-specs-tarball] 
		✔ [pull_request] 
		✔ [sha] 
	Target:
		✔ [agent-json-schema] APM agent json server schema 80b6af8d97e18344df38968bc5b9588961e89492

```

Reply:
but according to https://www.updatecli.io/docs/plugins/resource/shell/#_shell_condition:


> 💡 The source associated to this condition (the default source or through the key sourceID) is appended as the last argument to the command line unless the attribute disablesourceinput is set to true.

```
diff --git a/.ci/updatecli/updatecli.d/update-specs.yml b/.ci/updatecli/updatecli.d/update-specs.yml
index 0be8fa56a..2b3f34ef8 100644
--- a/.ci/updatecli/updatecli.d/update-specs.yml
+++ b/.ci/updatecli/updatecli.d/update-specs.yml
@@ -77,8 +77,8 @@ targets:
   agent-json-schema:
     name: APM agent json server schema {{ source "sha" }}
     scmid: default
-    sourceid: sha
+    disablesourceinput: true
     kind: shell
     spec:
-      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-schema.tgz && git --no-pager diff # required to ignore the sourceid being appended'
+      command: 'tar -xzf {{ requiredEnv "GITHUB_WORKSPACE" }}/json-schema.tgz && git --no-pager diff'
       workdir: "{{ .apm_agent.server_schema_specs_path  }}"
```

worked like a charm, so I missed the `disablesourceinput` argument when I tried this in the past. Let me add the change accordingly.

See 8e866912a

Reply:
Ah, right. `disablesourceinput` does the magic.